module/modmanager/manager_test.go

```diff
@@ -1430,3 +1431,125 @@ func TestModularDiscovery(t *testing.T) {
 		})
 	}
 }
+
+func TestFirstRun(t *testing.T) {
+	ctx := context.Background()
+	logger, logs := logging.NewObservedTestLogger(t)
+
+	exePath := rtestutils.BuildTempModuleWithFirstRun(t, "module/testmodule")
+	modCfg := config.Module{
+		Name:    "test-module",
+		ExePath: exePath,
+	}
+	parentAddr := setupSocketWithRobot(t)
+	opts := modmanageroptions.Options{
+		UntrustedEnv: false,
+	}
+	mgr := setupModManager(t, ctx, parentAddr, logger, opts)
+
+	// Subtests must be run sequentially in the given order.
+	// Re-ordering the tests is likely to cause failures.
```

Comment:
if the ordering matters, I don't want to make it a subtest. to me, subtests should work if run in parallel. I know this is not true generally, but want to start moving towards that model. Can we just do a t.Log() instead?

Reply:
that's a good point - I wanted to use sub-tests for organization+signposts, but logging banners or something works also - but i agree with you that it's better to create sub-tests that can be potentially be run in parallel.

the one minor thing we get from making sub-tests is cleanup (e.g. getting to unset an env var for free), but it's only an extra line or two to unset manually.